magma/array.py

```diff
@@ -679,10 +679,11 @@ def _make_t(self, index):
 
     def _resolve_slice_children(self, start, stop, slice_value):
         for i in range(start, stop):
-            if slice_value._ts:
+            if (not isinstance(slice_value.name, ArrayRef) or
+                    slice_value.name.array is not self):
                 self._ts[i] = slice_value[i - start]
             else:
-                slice_value._ts[i - start] = self[i]
+                slice_value._ts[i - start] = self._get_t(i)
```

Comment:
Call _get_t directly here to avoid the more general __getitem__ logic that's already being covered in the calling code path (don't need to call that code more than once)